src/client/BaseClient.js

```diff
@@ -43,19 +43,23 @@ class BaseClient {
     // Using Reduce to compute the validity of all extractors
     const allExtractorsValid = await this.extractors.reduce(async (curExtractorsValid, curExtractor) => {
       const { name } = curExtractor.constructor;
-
       if (curExtractor.validate) {
         logger.debug(`Validating ${name}`);
-        const isExtractorValid = await curExtractor.validate();
-        if (isExtractorValid) {
-          logger.debug(`Extractor ${name} PASSED CSV validation`);
-        } else {
+        try {
+          const isExtractorValid = await curExtractor.validate();
+          if (isExtractorValid) {
+            logger.debug(`Extractor ${name} PASSED CSV validation`);
+          } else {
+            logger.warn(`Extractor ${name} FAILED CSV validation`);
+          }
+          return ((await curExtractorsValid) && isExtractorValid);
+        } catch (e) {
           logger.warn(`Extractor ${name} FAILED CSV validation`);
```

Comment:
Two thoughts in one suggestion:

- With the log message here, we don't get any information about the error that gets thrown in the CSVURLModule. I think it would be nice to include that information. The first line in the suggestion is one way to address this.
- But then when you just do the first line, the generic error gets thrown on line 67 and it feels not ideal to lose the stack trace info of the real error. The second line to throw the error from the catch admittedly feels odd but also lets the final debug message have the stack track to the 404 error.

```suggestion
          logger.warn(`Extractor ${name} could not validate. Encountered the following error: ${e.message}`);
          throw e;
```